Vates/VatesAPI/src/vtkMDHistoQuadFactory.cpp

```diff
@@ -118,10 +119,12 @@ namespace Mantid
         is set so that all required vertices are marked, and created in a second step. */
 
         // Array of the points that should be created, set to false
-        bool * pointNeeded = new bool[nPointsX*nPointsY];
-        memset(pointNeeded, 0, nPointsX*nPointsY*sizeof(bool));
+        auto pointNeeded = Mantid::Kernel::make_unique<bool[]>(nPointsX * nPointsY);
+        for (int i = 0; i < nPointsX * nPointsY; i++) {
+          pointNeeded[i] = false;
+        }
```

Comment:
The four lines could be replaces by `std::vector<bool> pointNeded(nPointsX * nPointsY, false)`, I think.